### PR TITLE
Initial BoxedJob implementation

### PR DESCRIPTION
**Pull request by tardieu**
Implementation of a BoxedJob CRD and controller and kueue controller for BoxedJob objects.

### CODE REVIEW

**Comment by asm582** [comment]
Left some comments.

**Comment by tardieu** [comment]
> Left some comments.

Great! I added some clarifications.

File: config/rbac/role.yaml [hunk 1]
```diff
@@ -74,6 +82,48 @@ rules:
   - patch
   - update
   - watch
+- apiGroups:
```
Comment (dgrove-oss) [inline config/rbac/role.yaml:85]: I think we're going to have to either diverge from kubebuilder and spilt this into multiple files, or we're going to have to be smarter with kustomize patches and configs so we can avoid requiring Kueue-specific CRDs in the MCAD only configs (and core MCAD-specific CRDs in the Kueue config).

File: internal/controller/boxedjob_controller.go [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+	}
+	return nil
+}
+
+func parseComponent(job *workloadv1alpha1.BoxedJob, raw []byte) (*unstructured.Unstructured, error) {
+	obj := &unstructured.Unstructured{}
+	if _, _, err := unstructured.UnstructuredJSONScheme.Decode(raw, nil, obj); err != nil {
+		return nil, err
+	}
+	// TODO: use heuristic to inject labels
```
Comment (dgrove-oss) [inline internal/controller/boxedjob_controller.go:173]: I think with the change to `path` we don't need heuristics.  We can piggyback on the podset label injection we have to implement for kueue.

Comment (tardieu) [reply internal/controller/boxedjob_controller.go:173]: agreed

**Review by dgrove-oss — APPROVED** [review]
lgtm

**Pull request merged**